Dockerfile

```diff
@@ -1,30 +1,142 @@
+ENV HASHICORP_RELEASES=https://releases.hashicorp.com
+
+RUN set -eux && \
+    apk add --no-cache git bash wget openssl gnupg && \
+    gpg --keyserver keyserver.ubuntu.com --recv-keys C874011F0AB405110D02105534365D9472D7468F && \
+    mkdir -p /tmp/build && \
+    cd /tmp/build && \
+    apkArch="$(apk --print-arch)" && \
+    case "${apkArch}" in \
+        aarch64) packerArch='arm64' ;; \
+        armhf) packerArch='arm' ;; \
+        x86) packerArch='386' ;; \
+        x86_64) packerArch='amd64' ;; \
+        *) echo >&2 "error: unsupported architecture: ${apkArch} (see ${HASHICORP_RELEASES}/packer/${VERSION}/)" && exit 1 ;; \
+    esac && \
+    wget ${HASHICORP_RELEASES}/packer/${VERSION}/packer_${VERSION}_linux_${packerArch}.zip && \
+    wget ${HASHICORP_RELEASES}/packer/${VERSION}/packer_${VERSION}_SHA256SUMS && \
+    wget ${HASHICORP_RELEASES}/packer/${VERSION}/packer_${VERSION}_SHA256SUMS.sig && \
+    gpg --batch --verify packer_${VERSION}_SHA256SUMS.sig packer_${VERSION}_SHA256SUMS && \
+    grep packer_${VERSION}_linux_${packerArch}.zip packer_${VERSION}_SHA256SUMS | sha256sum -c && \
+    unzip -d /tmp/build packer_${VERSION}_linux_${packerArch}.zip && \
+    cp /tmp/build/packer /bin/packer && \
+    cd /tmp && \
+    rm -rf /tmp/build && \
+    gpgconf --kill all && \
+    apk del gnupg openssl && \
+    rm -rf /root/.gnupg && \
+    # Tiny smoke test to ensure the binary we downloaded runs
+    packer version
+
+ENTRYPOINT ["/bin/packer"]
+
+
+# Light development docker image which can be used to run the binary from a container.
+# This image builds from the locally generated binary in ./bin/, and from CI-built binaries within CI. 
+# To generate the local binary, run `make dev`.
+# This image is published to DockerHub under the `light`, `light-$VERSION`, and `latest` tags.
+FROM docker.mirror.hashicorp.services/alpine:latest as release-light
+
+ARG VERSION
+ARG BIN_NAME
+
+# TARGETARCH and TARGETOS are set automatically when --platform is provided.
+ARG TARGETOS TARGETARCH
+
+LABEL name="Packer" \
+      maintainer="HashiCorp Packer Team <packer@hashicorp.com>" \
+      vendor="HashiCorp" \
+      version=$VERSION \
+      release=$VERSION \
+      summary="Packer is a tool for creating identical machine images for multiple platforms from a single source configuration." \
+      description="Packer is a tool for creating identical machine images for multiple platforms from a single source configuration. Please submit issues to https://github.com/hashicorp/packer/issues"
+
+RUN apk add --no-cache git bash wget openssl gnupg
+
+COPY dist/$TARGETOS/$TARGETARCH/$BIN_NAME /bin/
+
+ENTRYPOINT ["/bin/packer"]
+
+
+# Full development docker image primarily used for development and debugging.
+# This image builds from source from the default branch of this repository. 
+# This image is published to DockerHub under the `full` and `full-$VERSION` tags.
+FROM docker.mirror.hashicorp.services/golang:alpine as release-full
```

Comment:
`full` is a weird docker target right now, and it's really not what we want.

In CI, the docker build action copies the CI-built binaries into the container under /bin/. E.g. the packer amd64 go binary built in github actions will be included in the docker amd64 image.. we **can** clone the packer repo and build from `master` within the dockerfile for this target, but we will be overwriting whatever has already been copied into /bin/. 

Looking through `scripts/build.sh`, I don't see anything being built other than the go binary for different goos/goarches, so I wonder if we want to use this at all. The docker build step will run when merges happen to `master`, so the binaries built and included in the docker image will always be the latest from `master`.. so my meta question is, do we need/want to continue supporting `full`?

Reply:
I'm removing the `full` target for now.